### ISSUE TITLE
Github action: Run OWASP ZAP Full Scan

### ISSUE THREAD

**Issue by rugk** [PrivateBin/PrivateBin#741]
I've discovered this GitHub action, and think it could be a good addition to just run a full automated vulnerability scan in our CI:
https://github.com/marketplace/actions/owasp-zap-full-scan

Done on `localhost` in the container, that should work, should not it?

**Edit:** Also, in addition to the DAST (=dynamic) code scanner above, there are a lot of static code scanners as well… I'll try setting up GitHub's own: https://github.com/PrivateBin/PrivateBin/security/code-scanning

**Comment by elrido** [comment]
OWASP ZAP - given this is based on a container, I tried it locally. It took 30 minutes to run, though it used barely RAM or CPU or network I/O - not sure what it did for so long - maybe waiting on timeouts? So, it's probably not appropriate to run on every push to master in this state. Anyhow, I collected the resulting output and you can review it at: https://snip.dssr.ch/?c53ba49d5dd2fdd3#2sAgiFS1MjVWYVgzLLzffuqJ6Svqiw5ZywZDHUex1rti
Let me know what you think about the result and if you have any actionable suggestions on that.

The codescanning stuff from Github links to some selected github actions. You can test them yourself by merging the action into master (and remove them again, if they send you weird mails or otherwise show odd behaviour). I think these two might be relevant to the project (links by github, leading you to a merge request and letting you review the action):
- GitHub CodeQL: https://github.com/PrivateBin/PrivateBin/new/master?filename=.github%2Fworkflows%2Fcodeql-analysis.yml&workflow_template=codeql
- Codacy (we used to run them, but their legacy integration started to give us trouble and you'd asked me to deactivate it): https://github.com/PrivateBin/PrivateBin/new/master?code_scanning_workflow_template=codacy&filename=.github%2Fworkflows%2Fcodacy-analysis.yml

PS: I see they offer Snyk (container package analyzer), will add that to our docker image repo.

**Comment by rugk** [comment]
>  So, it's probably not appropriate to run on every push to master in this state.

AFAIK it is by default configured to run once a day, IMHO this is an appropriate timing…

> Codacy (we used to run them, but their legacy integration started to give us trouble and you'd asked me to deactivate it): 

Yeah, they now have integration in this "GItHub Advanced Security" feature, so maybe it's worth to re-try…

**Comment by elrido** [comment]
I think this question has been answered and I am closing the issue. Please feel free to reply if you have follow-up questions or create a separate issue for different topics.

**Comment by rugk** [comment]
It has not, we'd still could add a YAML action/thing.

**Comment by rugk** [comment]
I guess it's easier/more useful to run it against a Docker image, so I opened a PR in https://github.com/PrivateBin/PrivateBin/issues/741. If you think it would be useful in the main repo here, though, we could of course move that here.

**Comment by rugk** [comment]
So, created PRs for both open tools.

**Comment by elrido** [comment]
As wished, I merged your PRs. Now that you have the scan results, what actions do you think we need to take to address them? Some of the reported issues may also be false positives.

**Comment by rugk** [comment]
I've excluded all false-positives in https://github.com/PrivateBin/docker-nginx-fpm-alpine/pull/30 for the OWASP scan.
All other issues in https://github.com/PrivateBin/docker-nginx-fpm-alpine/issues/29 are valid, but more of enhancements (stricter CSP etc.).

As far as I see the biggest risk both tools tell us is an outdated jQuery versions.

**Comment by elrido** [comment]
Once we got the bootstrap theme upgraded, maybe we should reconsider removing jquery in our code instead of upgrading it? The reason we were not able to upgrade it was that the old bootstrap failed with the later 3.x jQuery versions.

**Comment by rugk** [comment]
Sure, see #198 

Also #608
